### PR TITLE
fix: show error when format of exporter is not allowed

### PR DESCRIPTION
**Pull request by paolosantarsiero**
Bug: When try to export a file and the format is not allowed it return error 500.
Fix: Show a message that format is not allowed.

### CODE REVIEW

File: src/Controller/ExportController.php [hunk 1]
```diff
@@ -56,14 +56,19 @@ public function initialize(): void
...
-        }
+
         // check request (allowed methods and required parameters)
         $data = $this->checkRequest([
             'allowedMethods' => ['post'],
             'requiredParameters' => ['objectType'],
         ]);
+
+        if (!$this->Export->checkFormat($format)) {
+            $this->Flash->error('Il formato selezionato non è disponibile!');
```
Comment (stefanorosanelli) [inline src/Controller/ExportController.php]: We use english for labels and messages with `__()` function to enable translations via gettext
It should be:

```php
$this->Flash->error(_('Selected format....'));
```

File: src/Controller/ExportController.php [hunk 2]
```diff
@@ -56,14 +56,19 @@ public function initialize(): void
...
+
+            return $this->redirect(['_name' => 'modules:list', 'object_type' => $data['objectType']]);
```
Comment (stefanorosanelli) [inline src/Controller/ExportController.php]: Is there a test case for these lines?

**Review by stefanorosanelli — CHANGES_REQUESTED** [review]

File: composer.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
         "josegonzalez/dotenv": "2.*",
         "mobiledetect/mobiledetectlib": "2.*",
         "phpoffice/phpspreadsheet": "^1.15",
-        "wikimedia/composer-merge-plugin": "^1.4"
+        "wikimedia/composer-merge-plugin": "^2.0"
```
Comment (didoda) [inline composer.json]: This change is not related to the fix (it's related to the use of composer2)

File: src/Controller/ExportController.php [hunk 4]
```diff
@@ -56,14 +56,19 @@ public function initialize(): void
...
         // check request (allowed methods and required parameters)
         $data = $this->checkRequest([
             'allowedMethods' => ['post'],
             'requiredParameters' => ['objectType'],
         ]);
+
+        if (!$this->Export->checkFormat($format)) {
+            $this->Flash->error(__('Format not allowed'));
+
+            return $this->redirect(['_name' => 'modules:list', 'object_type' => $data['objectType']]);
```
Comment (stefanorosanelli) [inline src/Controller/ExportController.php]: safer redirect is probably this (we can't be sure where the request originated)

```php
return $this->redirect($this->referer());
```

File: composer.json (hunk 3 above)
Comment (stefanorosanelli) [reply composer.json]: right, but why do we have this diff since in `master` we have already `"wikimedia/composer-merge-plugin": "^2.0.1"` ? :thinking:

Comment (didoda) [reply composer.json]: some merge conflict. resolved with a push...

**Review by didoda — APPROVED** [review]

File: tests/TestCase/Controller/ExportControllerTest.php [hunk 5]
```diff
@@ -156,6 +156,28 @@ public function testExport(): void
...
+        $this->Export = new ExportController(
+            new ServerRequest([
+                'environment' => ['REQUEST_METHOD' => 'POST'],
+                'params' => ['objectType' => 'proms'],
+                'post' => ['ids' => '655', 'objectType' => 'proms', 'format' => ''],
+            ])
+        );
+
+        // call export.
+        $response = $this->Export->export();
```
Comment (stefanorosanelli) [inline tests/TestCase/Controller/ExportControllerTest.php:176]: [minor]
```
static::assertEquals(302, $response->getStatusCode());
```

File: src/Locale/en_US/default.po [hunk 6]
```diff
@@ -191,6 +191,9 @@ msgstr ""
 msgid "Export Filtered"
 msgstr ""
 
+msgid "Format not allowed"
+msgstr "Format choosen is not available"
```
Comment (stefanorosanelli) [inline src/Locale/en_US/default.po]: We usually don't _translate_ messages in this locale

use `"Format choosen is not available"` directly as message string instead of `"Format not allowed"` and update references

**Review by stefanorosanelli — APPROVED** [review]
